Updating docs to include new assume role fields (#29649)
hashicorp/vault: website/content/api-docs/secret/aws.mdx

with an AWS IAM User, which will be adopted and managed by Vault, including rotating it according
to the configured `rotation_period`.

> In addition, this endpoint (Requires Vault Enterprise 1.19+) now supports cross account management.

Reviewer: ```suggestion
In addition, this endpoint supports cross account management.
```

I believe the recommendation from the Education team is to omit references like this since we have versioned docs.

> In addition, this endpoint supports cross account management.
> Vault will use the credentials obtained by assuming roles in another AWS account to perform AWS operations.

Reviewer 2: ```suggestion
In addition, this endpoint supports cross-account management.
Vault will use the credentials obtained by assuming a role in another AWS account to perform AWS operations.
```
I think it should be singular because a static role is only configured to assume 1 role each.

> Make sure that the IAM user or role configured in Vault has the necessary permissions to assume those roles.

Reviewer 2: ```suggestion
Make sure that the IAM role configured in Vault has the necessary permissions to manage the IAM user within the target account.
```
Please let me know if I'm mistaken, but I think this is the information you are trying to say? I think the current wording is a bit confusing.

specified in either `24h` or `86400` format (see [duration format strings](/vault/docs/concepts/duration-format)).
Updating the rotation period will 'reset' the next rotation to occur at `now` + `rotation_period`.

- `assume_role_arn` `(string)` – Specifies the ARN of the role that Vault should assume.
When provided, Vault will use AWS STS to assume this role and generate temporary credentials.
If `assume_role_arn` is provided, `assume_role_session_name` must also be provided.

- `assume_role_session_name` `(string)` – Specifies the session name to use when assuming the role.
If `assume_role_session_name` is provided, `assume_role_arn` must also be provided.

> - `external_id` `(string)` – Specifies the external ID to use when assuming the role.

Author: For the Enterprise tag, I know that we want to omit using ent references, but I am guessing that's only for the versions, and tags should be good? https://github.com/hashicorp/vault/pull/29649#discussion_r1984074968

@hashicorp/vault-education-approvers any input?

### Sample response

```json
{
  "data": {
    "assume_role_arn": "",
    "assume_role_session_name": "",
    "external_id": "",

>     "id": "AIDA..",

Reviewer 2: ```suggestion
    "id": "AIDA...",
```
Super nitpick lol.

    "name": "my-static-role",
    "rotation_period": 41400,
    "username": "example-user"
  }
}
```

> 

Reviewer: 👍
